src/main/java/uk/gov/hmcts/reform/pip/data/management/service/PublicationService.java

```diff
@@ -32,25 +35,67 @@ public PublicationService(ArtefactRepository artefactRepository,
+     *
      * @param artefact The artifact that needs to be created.
-     * @param payload The paylaod for the artefact that needs to be created.
+     * @param payload  The payload for the artefact that needs to be created.
      * @return Returns the UUID of the artefact that was created.
      */
     public Artefact createPublication(Artefact artefact, String payload) {
 
-        Optional<Artefact> foundArtefact =  artefactRepository
+        Optional<Artefact> foundArtefact = artefactRepository
             .findBySourceArtefactIdAndProvenance(artefact.getSourceArtefactId(), artefact.getProvenance());
 
         foundArtefact.ifPresent(value -> artefact.setArtefactId(value.getArtefactId()));
 
         String blobUrl = azureBlobService.createPayload(
             artefact.getSourceArtefactId(),
             artefact.getProvenance(),
-            payload);
+            payload
+        );
 
         artefact.setPayload(blobUrl);
         artefact.setSearch(payloadExtractor.extractSearchTerms(payload));
 
         return artefactRepository.save(artefact);
     }
+
+
+    /**
+     * Get all relevant artefacts relating to a given court ID.
+     *
+     * @param searchValue - represents the court ID in question being searched for
+     * @param verified    - represents the verification status of the user. Currently only verified/non-verified, but
+     *                    will include other verified user types in the future
+     * @return a list of all artefacts that fulfil the timing criteria, match the given court id and sensitivity
+     *                    associated with given verification status
+     */
+    public List<Artefact> findAllByCourtId(String searchValue, Boolean verified) {
+        LocalDateTime currDate = LocalDateTime.now();
+        if (verified) {
+            return artefactRepository.findArtefactsBySearchVerified(searchValue, currDate);
+        } else {
+            return artefactRepository.findArtefactsBySearchUnverified(searchValue, currDate);
+        }
+
+    }
+
+    /**
+     * takes in provenance and artefact id and returns the data within the matching blob in string format.
+     * @param artefactId represents the artefact id which is then used to get an artefact to populate the inputs
+     *                   for the blob request
+     * @return the data within the blob in string format
+     */
+    public String getByArtefactId(UUID artefactId) {
+        Optional<Artefact> optionalArtefact = artefactRepository.findByArtefactId(artefactId);
+        if (optionalArtefact.isPresent()) {
+            Artefact artefact;
+            artefact = optionalArtefact.get();
+            String sourceArtefactId = artefact.getSourceArtefactId();
+            String provenance = artefact.getProvenance();
+            return azureBlobService.getBlobData(sourceArtefactId, provenance);
+        } else {
+            return azureBlobService.getBlobData("", "");
```

Comment:
Is this needed / can we just throw an exception here instead?

Reply:
To be honest, I just did this so as not to break the existing tests. Absolutely can just throw an exception, yeah :)